From 40a157d2161fdd8709a777de6625fbd1b93efed3 Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Sat, 6 Aug 2016 19:23:48 -0400
Subject: [PATCH 1/1] src/skyutils.h: fix two missing format strings in
 SU_SYSLOG_FN.

The SU_SYSLOG_FN(x,y) macro could be defined as either syslog(x,y) or
printf(y), both of which are missing the trivial format string
"%s". This commit adds it, so that the project builds with no
-Wformat-security warnings.

Gentoo-Bug: 521114
---
 src/skyutils.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/skyutils.h b/src/skyutils.h
index 8b7d59d..f77c9fd 100644
--- a/src/skyutils.h
+++ b/src/skyutils.h
@@ -279,14 +279,14 @@ void SU_PrintDebug(int Level,char *Txt, ...);
 #ifdef __unix__
 #define SU_PrintDebug(x,...) /* If you have an error here, remove 'x,...' from the () */
 #include <syslog.h>
-#define SU_SYSLOG_FN(x,y) syslog(x,y)
+#define SU_SYSLOG_FN(x,y) syslog(x,"%s",y)
 #else /* !__unix__ */
 #define SU_PrintDebug() /* Avoid compilation warnings */
 extern FILE *SU_LogFile;
 #define SU_SYSLOG_FN(x,y) SU_WriteToLogFile(SU_LogFile,y)
 #endif /* __unix__ */
 #else /* DEBUG */
-#define SU_SYSLOG_FN(x,y) printf(y)
+#define SU_SYSLOG_FN(x,y) printf("%s",y)
 #endif /* !DEBUG */
 #ifdef _WIN32
 #define SU_SLEEP(x) Sleep(x*1000)
-- 
2.7.3

